stdlib/src/builtin/sort.mojo

```diff
@@ -15,400 +15,169 @@
+        list: The list of the scalars which will be sorted inpace.
     """
-    # Downcast any pointer to register-passable pointer.
-    var ptr = rebind[Pointer[Int]](v.data)
-    sort(ptr, len(v))
+    var count = len(list)
+    if count <= 64:
+        insertion_sort(list)  # small lists are best sorted with insertion sort
+    else:
+        quick_sort(list)  # medium lists are best sorted with quick sort
 
 
-fn sort[type: DType](inout v: List[Scalar[type]]):
-    """Sort the vector inplace.
-    The function doesn't return anything, the vector is updated inplace.
+# ===----------------------------------------------------------------------=== #
+#  Comparable elements list sorting
+# ===----------------------------------------------------------------------=== #
+
+
+@always_inline
+fn insertion_sort[D: ComparableCollectionElement](inout list: List[D]):
+    """Sort list of the order comparable elements in place with insertion sort algorithm.
 
     Parameters:
-        type: DType of the underlying data.
+        D: The order comparable collection element type.
 
     Args:
-        v: Input vector to sort.
+        list: The list of the order comparable elements which will be sorted inpace.
     """
-
-    var ptr = rebind[Pointer[Scalar[type]]](v.data)
-    sort[type](ptr, len(v))
+    for i in range(1, len(list)):
+        var key = list[i]
+        var j = i - 1
+        while j >= 0 and key < list[j]:
+            list[j + 1] = list[j]
+            j -= 1
+        list[j + 1] = key
 
 
-fn sort[
-    type: CollectionElement,
-    cmp_fn: fn (type, type) capturing -> Bool,
-](inout v: List[type]):
-    """Sort the vector inplace.
-    The function doesn't return anything, the vector is updated inplace.
+fn _quick_sort[
+    D: ComparableCollectionElement
+](inout list: List[D], low: Int, high: Int):
+    """Sort section of the list, between low and high, with quick sort algorithm inplace.
 
     Parameters:
-        type: DType of the underlying data.
-        cmp_fn: The comparison function.
+        D: The order comparable collection element type.
 
     Args:
-        v: Input vector to sort.
+        list: The list of the order comparable elements which will be sorted inpace.
+        low: Int value identifying the lowes index of the list section to be sorted.
+        high: Int value identifying the highest index of the list section to be sorted.
     """
 
-    _quicksort[type, cmp_fn](v.data, len(v))
-
-
-# ===----------------------------------------------------------------------===#
-# sort networks
-# ===----------------------------------------------------------------------===#
+    @always_inline
+    @parameter
+    fn _partition(low: Int, high: Int) -> Int:
+        var pivot = list[high]
+        var i = low - 1
+        for j in range(low, high):
+            if list[j] <= pivot:
+                i += 1
+                list[j], list[i] = list[i], list[j]
+        list[i + 1], list[high] = list[high], list[i + 1]
```

Comment:
**Suggestion** Split the assignment across two lines please for readability.

```mojo
list[i+1] = list[high]
list[high] = list[i+1]
```

Reply:
We would need to introduce a temp var and have a third statement:
```Mojo
var tmp = list[i+1]
list[i+1] = list[high]
list[high] = tmp
```

It is simpler to read but not "pythonic" :)